Source Code in txt/RPi/image_processing_server/config.py.txt
MODEL_NAME = 'model'
INFERENCE_GRAPH = 'frozen_inference_graph.pb'
LABEL_MAP = 'labelmap.pbtxt'

# Number of classes the object detector can identify
NUM_CLASSES = 15

IMAGE_ENCODING = '.png'
STOPPING_IMAGE = 'stop_image_processing.png'

MAX_NUM_SYMBOLS = 3

IMAGE_WIDTH = 1920  # 400
IMAGE_HEIGHT = 1080  # 225

DISPLAY_IMAGE_WIDTH = 400

# red colour symbols tend to have lower confidence scores
MIN_CONFIDENCE_THRESHOLD = 0.50

# usually for non-red symbols, confidence of > 90%.
# however, once in a blue moon, confidence score may drop very low.
# no false positive with confidence higher than 70% though
# therefore, set confidence score this low
NON_RED_CONFIDENCE_THRESHOLD = 0.70

# used for filtering symbols that are 5 grids away
# sitution: [S]    [ ]  <R
# where [ ] be obstacle,
#       S be symbol
#       R be robot
#       < be camera direction
# 3 grids - extreme case (correct): ~750
# 5 grids (wrong; as shown in situation): ~695
YMAX_THRESHOLD = 775

SYMBOL_ON_LEFT_OF_IMAGE_THRESHOLD = 780  # left xmax compared to middle xmin
SYMBOL_ON_RIGHT_OF_IMAGE_THRESHOLD = 1090  # right xmin compared to middle xmax

MAIN_IMAGE_DIR = 'frames'
RAW_IMAGE_DIR = 'raw'
PROCESSED_IMAGE_DIR = 'processed'

RAW_IMAGE_PREFIX = 'frame'
PROCESSED_IMAGE_PREFIX = 'processed'

DISPLAY_DURATION_MILLISECONDS = 3000

LEFT_OBSTACLE = 'left_obstacle'
MIDDLE_OBSTACLE = 'middle_obstacle'
RIGHT_OBSTACLE = 'right_obstacle'

NO_SYMBOL = '-1'


Source Code in txt/MDP-Android/app/src/main/java/com/example/mdp_android_grp15/ui/main/ReconfigureFragment.java.txt
package com.example.mdp_android_grp15.ui.main;

import android.app.DialogFragment;
import android.content.Context;
import android.content.DialogInterface;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.WindowManager;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.annotation.Nullable;

import com.example.mdp_android_grp15.MainActivity;
import com.example.mdp_android_grp15.R;

public class ReconfigureFragment extends DialogFragment {
    private static final String TAG = "ReconfigureFragment";
    SharedPreferences sharedPreferences;
    SharedPreferences.Editor editor;

    Button saveBtn, cancelReconfigureBtn;
    EditText f1ValueEditText, f2ValueEditText;
    String f1Value, f2Value;
    View rootView;
    View commsView;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, Bundle savedInstanceState) {
        showLog("Entering onCreateView");
        rootView = inflater.inflate(R.layout.activity_reconfigure, container, false);
        commsView = inflater.inflate(R.layout.activity_comms, container, false);

        super.onCreate(savedInstanceState);

        getDialog().setTitle("Reconfiguration");

        saveBtn = rootView.findViewById(R.id.saveBtn);
        cancelReconfigureBtn = rootView.findViewById(R.id.cancelReconfigureBtn);
        f1ValueEditText = rootView.findViewById(R.id.f1ValueEditText);
        f2ValueEditText = rootView.findViewById(R.id.f2ValueEditText);

        sharedPreferences = getActivity().getSharedPreferences("Shared Preferences", Context.MODE_PRIVATE);

        if (sharedPreferences.contains("F1")) {
            f1ValueEditText.setText(sharedPreferences.getString("F1", ""));
            f1Value = sharedPreferences.getString("F1", "");
            getDialog().getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_ALWAYS_HIDDEN);

        }
        if (sharedPreferences.contains("F2")) {
            f2ValueEditText.setText(sharedPreferences.getString("F2", ""));
            f2Value = sharedPreferences.getString("F2", "");
        }

        if (savedInstanceState != null) {
            f1Value = savedInstanceState.getStringArray("F1F2 value")[0];
            f2Value = savedInstanceState.getStringArray("F1F2 value")[1];
        }

        saveBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                showLog("Clicked saveBtn");
                editor = sharedPreferences.edit();
                editor.putString("F1", f1ValueEditText.getText().toString());
                editor.putString("F2", f2ValueEditText.getText().toString());
                editor.commit();
                if (!sharedPreferences.getString("F1", "").equals(""))
                    f1Value = f1ValueEditText.getText().toString();
                if (!sharedPreferences.getString("F2", "").equals(""))
                    f2Value = f2ValueEditText.getText().toString();
                Toast.makeText(getActivity(), "Saving values...", Toast.LENGTH_SHORT).show();

                showLog("Exiting saveBtn");
                getDialog().dismiss();
            }
        });

        cancelReconfigureBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                showLog("Clicked cancel ReconfigureationBtn");
                if (sharedPreferences.contains("F1"))
                    f1ValueEditText.setText(sharedPreferences.getString("F1", ""));
                if (sharedPreferences.contains("F2"))
                    f2ValueEditText.setText(sharedPreferences.getString("F2", ""));
                showLog("Exiting cancel ReconfigureationBtn");
                getDialog().dismiss();
            }
        });
        showLog("Exiting onCreateView");
        return rootView;
    }

    @Override
    public void onDismiss(DialogInterface dialog) {
        showLog("Entering onDismiss");
        super.onDismiss(dialog);
        if (f1Value != null && !f1Value.equals(""))
            MainActivity.getF1().setContentDescription(f1Value);
        if (f2Value != null && !f2Value.equals(""))
            MainActivity.getF2().setContentDescription(f2Value);
        f1ValueEditText.clearFocus();

        showLog("Exiting onDismiss");
    }

    @Override
    public void onSaveInstanceState(Bundle outState) {
        showLog("Entering onSaveInstanceState");
        super.onSaveInstanceState(outState);

        String[] value = new String[]{f1Value, f2Value};
        showLog("Exiting onSaveInstanceState");
        outState.putStringArray(TAG, value);
    }

    private void showLog(String message) {
        Log.d(TAG, message);
    }
}


RPi/rpi/src/config.py
LOCALE = 'UTF-8'

# Android BT connection settings
RFCOMM_CHANNEL = 9
RPI_MAC_ADDR = 'B8:27:EB:99:A8:38'
UUID = '443559ba-b80f-4fb6-99d9-ddbcd6138fbd'
ANDROID_SOCKET_BUFFER_SIZE = 512

# Algorithm Wifi connection settings
# raspberryHotPotato: 192.168.3.1
WIFI_IP = '192.168.15.15'
WIFI_PORT = 8080
ALGORITHM_SOCKET_BUFFER_SIZE = 512

# Arduino USB connection settings
# SERIAL_PORT = '/dev/ttyACM0'
# Symbolic link to always point to the correct port that arduino is connected to
SERIAL_PORT = '/dev/serial/by-id/usb-Arduino__www.arduino.cc__0043_75232303235351F091C0-if00'
BAUD_RATE = 57600

# Image Recognition Settings
STOPPING_IMAGE = 'stop_image_processing.png'

IMAGE_WIDTH = 1920
IMAGE_HEIGHT = 1080
IMAGE_FORMAT = 'bgr'

BASE_IP = 'tcp://192.168.15.'
PORT = ':5555'

IMAGE_PROCESSING_SERVER_URLS = {
    'cheyanne': BASE_IP + '54' + PORT,
    'elbert': BASE_IP + '00' + PORT,  # don't have elbert's ip address yet
    'jason': BASE_IP + '52' + PORT,
    'joshua': BASE_IP + '93' + PORT,
    'mingyang': BASE_IP + '74' + PORT,
    'reuben': BASE_IP + '00' + PORT,  # don't have reuben's ip address yet
    'winston': BASE_IP + '55' + PORT,
    'yingting': BASE_IP + '90' + PORT,
}
